commands/publish/lib/git-commit.js

```diff
@@ -7,11 +7,13 @@ const childProcess = require("@lerna/child-process");
 
 module.exports = gitCommit;
 
-function gitCommit(message, opts) {
+function gitCommit(message, amend, opts) {
```

Comment:
I'm not a huge fan of this extra argument (that is _usually_ `false`). Can we move this to the `opts` object instead?

`opts` is just `this.execOpts` from the PublishCommand instance. In the constructor, when `this.options.amend === true`, we can set `this.execOpts.amend = true`. We want to avoid adding the execOpts prop when the option is falsey because otherwise the tests are a bear to update.